### PR TITLE
Avoid 10 seconds delay on admin-tools container termination

### PR DESCRIPTION
**Pull request by mjameswh**
## What was changed
- Added `tini` to admin-tools's entrypoint

## Why?
- Avoid 10 seconds delay on admin-tools container termination, due to incorrect SIGTERM handling by busybox's `tail` command (see below image for behavior before this PR)

Previous termination behaviour

More context
```
This is likely the result of the PID 1 signaling issue.

Inside a PID namespace the first process launched gets PID 1.
PID 1 is treated special by the kernel in that the default signal actions are to ignore, instead of exit.
Meaning that unless the application installs a signal handler,
SIGINT and SIGTERM won't cause the application to exit. (see http://lwn.net/Articles/532748/)

tail is a very basic utility, and does not install any signal handlers, so it ignores them.
The only thing that will kill it is SIGKILL (which is what docker stop will do after the 10 second timeout)
```

## Checklist
1. How was this tested: built and ran affected image locally

2. Any docs updates needed? No.

### CODE REVIEW

**Comment by mjameswh** [comment]
**Notes to reviewer**

I wasn't sure about expectations for PR on this repo.
- This PR is targeted at branch 1.18.x, but the issue also applies to other branch. Should I target 'main' instead?
- In admin-tools.Dockerfile, `BASE_ADMIN_TOOLS_IMAGE`'s version will need to be bumped. Who updates that version number?

**Comment by mindaugasrukas** [comment]
> **Notes to reviewer**
> 
> I wasn't sure about expectations for PR on this repo.
> 
> * This PR is targeted at branch 1.18.x, but the issue also applies to other branch. Should I target 'main' instead?
> * In admin-tools.Dockerfile, `BASE_ADMIN_TOOLS_IMAGE`'s version will need to be bumped. Who updates that version number?

All PRs should go against the `main` branch. We will cherrypick changes to release branches if necessary and update versions during the release process.

**Comment by mindaugasrukas** [comment]
Added more context to the issue.

File: docker/base-images/base-admin-tools.Dockerfile [hunk 1]
```diff
@@ -9,10 +9,11 @@ RUN apk add --update --no-cache \
     mysql-client \
     postgresql-client \
     py-pip \
+    tini \
     && pip install cqlsh
 
 # set up nsswitch.conf for Go's "netgo" implementation
 # https://github.com/gliderlabs/docker-alpine/issues/367#issuecomment-424546457
-RUN test ! -e /etc/nsswitch.conf && echo 'hosts: files dns' > /etc/nsswitch.conf
+RUN test ! -e /etc/nsswitch.conf && echo 'hosts: files dns' > /etc/nsswitch.conf || true
```
Comment (mindaugasrukas) [inline docker/base-images/base-admin-tools.Dockerfile:17]: Any context why this change?

Comment (mjameswh) [reply docker/base-images/base-admin-tools.Dockerfile:17]: If the `test` command returns false, then the whole `RUN` command evaluate to `false` and the docker build fails because of that.

I have no idea why this error actually happened on my local build and not to anyone before, but for sure, either that command should not be conditional or it should have the `|| true` at the end...

**Comment by mindaugasrukas** [comment]
@mjameswh I changed the PR base to `main`. Could you rebase your PR to resolve conflicts?

**Comment by mjameswh** [comment]
@mindaugasrukas I rebased on branch main. Sorry for the delay.

File: docker/base-images/base-admin-tools.Dockerfile (hunk 1 above)
Comment (robholland) [reply docker/base-images/base-admin-tools.Dockerfile:18]:
```suggestion
RUN test -e /etc/nsswitch.conf || echo 'hosts: files dns' > /etc/nsswitch.conf
```

File: admin-tools.Dockerfile [hunk 2]
```diff
@@ -46,4 +46,4 @@ COPY --from=admin-tools-builder /home/builder/temporal/schema /etc/temporal/sche
 COPY --from=admin-tools-builder /home/builder/temporal/tdbg /usr/local/bin
 
 # Keep the container running.
-ENTRYPOINT ["tail", "-f", "/dev/null"]
+ENTRYPOINT ["tini", "--", "tail", "-f", "/dev/null"]
```
Comment (robholland) [inline admin-tools.Dockerfile:49]: Unrelated to this PR, but while you're here...

```suggestion
ENTRYPOINT ["tini", "--", "sleep", "infinity"]
```

That might be a bit more intuitive that we're trying to deliberately keep the container alive. Feel free to ignore this though, just an aside.

Comment (mindaugasrukas) [reply admin-tools.Dockerfile]: TIL.

However, I found this comment on StackOverflow:
```
Regarding this, I made some research I documented in a separate answer. To summarize: infinity is converted in C from "string" to a double. Then that double is truncated to the maximum values allowed timespec, which means a very large amount of seconds (architecture-dependant) but, in theory, finite.
```

**Review by mindaugasrukas — APPROVED** [review]
